doc/synchronous-mode/synchronous-mode-cfg.md

```diff
@@ -0,0 +1,40 @@
+# 1. Scope
+This document describes the design for the configuration of synchronous mode between orchagent and syncd.
+
+# 2. A brief of synchronous mode
+An illustrative figure of the synchronous mode is shown below. The synchronous mode achieves a closed-loop execution if SAI APIs from orchagent. In contrast to the previous asynchronous mode which cannot properly handle SAI API failures, the synchronous mode can gracefully handle SAI API failures by conducting the proper actions in orchagent. Therefore, the synchronous mode can substantially improve the reliability of SONiC.
+
+<img src="synchronous-mode-diagram.png" width="500">
+
+# 3. Configuration design 
+### 3.1 Configuration Requirements
+1.	Allow users to enable or disable synchronous mode via CLI.
+2.	Follow the image-specified mode when upgrading an image with cold/warm/fast-reboot if the synchronous mode configuration is not explicitly specified by users in configDB.
+3.	Use the user-specified mode when an explicit configuration of synchronous mode exists in configDB.
+4.	Support enabling/disabling the synchronous mode when deploying a minigraph.
+
+### 3.2 ConfigDB changes
+The following field is added to `DEVICE_METADATA|localhost` in configDB to support the configuration of synchronous mode
+```
+DEVICE_METADATA|localhost
+    "synchronous_mode": "enable|disable"
+```
+There are three possibilities for the field with the behavior as follows:
+| Field value               | Behavior                          |
+|---------------------------|-----------------------------------|
+| enable                    | Enable synchronous mode.           |
+| disable                   | Disable synchronous mode.          |
+| Field does not exist      | Orchagent and syncd use the mode specified by `orchagent.sh` and `syncd-init-common.sh`, respectively. |

```

Comment:
>Field does not exist [](start = 2, length = 20)

Let's be verbose. If the field does not exist, and user never use the CLI, will it appear in ConfigDB or not at all?

Reply:
Added a description for this scenario.